python/src/com/google/idea/blaze/python/resolve/provider/BazelPyImportResolverStrategy.java

```diff
@@ -51,6 +51,15 @@ protected QualifiedName toImportString(ArtifactLocation source) {
     if (source.isGenerated() || !source.getRelativePath().endsWith(".py")) {
       return null;
     }
-    return fromRelativePath(source.getRelativePath());
+    QualifiedName name = fromRelativePath(source.getRelativePath());
+    // pip_parse and pip_install macros unzip pip wheels into site-packages dir
```

Comment:
I'm not familiar with the issue, but in this case, wouldn't it be more "correct" to get the relative path with `site-packages` as root?

Reply:
The issue is this one: https://github.com/bazelbuild/intellij/issues/2239

I do not know structure of the plug-in, this solution is very likely hacky. The gist of it:
- py_install, py_parse bazel macros bring pip packages under "external/repo_package/site-packages" - this doesn't work with IntelliJ
- http_archive bazel macro, on the other hand, brings pip dependencies (from tar balls) to "external/repo_package" - this works fine with IntelliJ

py_install and py_parse are much more convenient because one can specify requirements.txt with a list of pip packages to bring. I narrowed it down to this function that converts relative python module paths to import strings. Given it was my first time looking at the plug-in, likely the right fix is entirely different - say in py_parse/py_install macros or elsewhere in the plugin.

What would you advise we should do here? Can bazel plug-in owners take this issue on - repro and resolve in a proper way?

Reply:
@jastice what would you suggest we do here?

Here is an example py_binary - snippet of bazel-generated Main function that shows how imports are generated for pip packages. Notice **site-packages** prefixes in python imports. That is how bazel understands the import statements without site-packages, but InteliJ bazel plug-in does not understand them because imports are generated with site-packages in them. The fix in this commit, removes site-packages from the potential import statements.

```python
def Main():
  args = sys.argv[1:]

  new_env = {}

  if IsRunningFromZip():
    module_space = CreateModuleSpace()
  else:
    module_space = FindModuleSpace()

  python_imports = 'pip_cloudpickle/site-packages:pip_crcmod/site-packages:pip_dill/site-packages:pip_fastavro/site-packages:pip_six/site-packages:pip_grpcio/site-packages:pip_docopt/site-packages:pip_certifi/site-packages:pip_charset_normalizer/site-packages:pip_idna/site-packages:pip_urllib3/site-packages:pip_requests/site-packages:pip_hdfs/site-packages:pip_pyparsing/site-packages:pip_httplib2/site-packages:pip_numpy/site-packages:pip_orjson/site-packages:pip_protobuf/site-packages:pip_proto_plus/site-packages:pip_pyarrow/site-packages:pip_pydot/site-packages:pip_pymongo/site-packages:pip_python_dateutil/site-packages:pip_pytz/site-packages:pip_typing_extensions/site-packages:pip_apache_beam/site-packages:com_github_protocolbuffers_protobuf/python:pip_absl_py/site-packages:pip_setuptools/site-packages'
  python_path_entries = CreatePythonPathEntries(python_imports, module_space)
  python_path_entries += GetRepositoriesImports(module_space, True)
  # Remove duplicates to avoid overly long PYTHONPATH (#10977). Preserve order,
  # keep first occurrence only.
```

Reply:
@tpasternak @mai93 @jastice - please approve or advise how to move this forward. Without the fix, updates to IntelliJ force us to rebuild the plug-in with this change.